src/user/rust.rs

```diff
@@ -0,0 +1,198 @@
+use parse_display::{Display, FromStr};
+use rust_nix_templater::{options::RustToolchainChannel, *};
+
+use crate::ActionStack;
+
+use super::{UserAction, UserMetadata, UserPrompt};
+
+#[derive(Debug, Clone, PartialEq, Eq, Display, FromStr)]
+pub(crate) enum Prompt {
+    #[display("generate flake")]
+    Generate,
+    #[display("set package name ({0})")]
+    SetPackageName(String),
+    #[display("set executable name ({0})")]
+    SetExecName(String),
+    #[display("set description ({0})")]
+    SetDescription(String),
+    #[display("set toolchain ({0})")]
+    SetToolchain(RustToolchainChannel),
+    #[display("set license ({0})")]
+    SetLicense(String),
+    #[display("toggle build outputs ({0})")]
+    ToggleBuildOutputs(bool),
+    #[display("toggle app outputs ({0})")]
+    ToggleAppOutputs(bool),
+    #[display("toggle library flag ({0})")]
+    ToggleLibrary(bool),
+    #[display("{0} channel")]
+    ChooseToolchain(RustToolchainChannel),
+    #[display("{0} license")]
+    ChooseLicense(String),
```

Comment:
we prefill this usually, no need to always allocate.
```suggestion
    ChooseLicense(std::borrow::Cow<'static, str>),
```

Reply:
`Cow<'static, str>` doesn't implement `FromStr`, perhaps we could pull in https://crates.io/crates/smol_str?

Reply:
ok, sounds good.